### ISSUE TITLE
GNUPGHOME isn't respected

### ISSUE THREAD

**Issue by jasonkarns** [keybase/keybase-issues#2178]
I have `GNUPGHOME` set to a directory under `XDG_CONFIG_HOME`. However, any of the keybase pgp subcommands are still expecting/recreating `~/.gnupg` rather than respecting `GNUPGHOME`.

```
$ keybase --version
keybase version 1.0.14-1
```

**Comment by maxtaco** [comment]
this might be an issue of a restart of the service needed before the environment variable is sourced

**Comment by jasonkarns** [comment]
I restarted the service, but it still seems to only be writing to .gnupg. (I'm basing this off the fact that `keybase pgp pull` regenerates the .gnupg directory).

**Comment by jasonkarns** [comment]
FWIW, I set up a symlink for .gnupg for now, but I would love to figure this out.

**Comment by maxtaco** [comment]
And the service now has the GNUPGHOME variable set?  You can ask `ps` to tell you the environment of the service.

**Comment by jasonkarns** [comment]
It appears not to. I have `GNUPGHOME` exported in `.bashrc`, which is in turn sourced by `.profile`.

How is this expected to be set? `launchctl`? Should it be launched by a different user?

**Comment by maxtaco** [comment]
Yeah, `launchctl setenv` should do it.  Annoying problem, sorry about that....

**Comment by jasonkarns** [comment]
Thanks!

Can a feature request be opened so this isn't necessary? I fully expect to
forget about this and be baffled in 6mo when I get a new computer or
reinstall, you know?

Thanks again for the help.

**Comment by jasonkarns** [comment]
Followup comment as I just realized this will be doubling annoying if my XDG_CONFIG_HOME ever changes. My GNUPGHOME is set as: `export GNUPGHOME=$XDG_CONFIG_HOME/gpg` but launchctl won't have the proper XDG_CONFIG_HOME, so I'll have to duplicate/hard-code that.

**Comment by jasonkarns** [comment]
Planning on submitting a PR to the homebrew formula to include this as a caveat:

https://github.com/Homebrew/homebrew-core/commit/efab5ca281b590ad2f6ca8ec90a903d08997cebe

thoughts?
